src/sagemaker/jumpstart/types.py

```diff
@@ -1001,6 +1007,206 @@ def __init__(
+            json_obj (Dict[str, Any]): Dictionary representation of hub content description.
+        """
+
+        self.dependency_copy_path: Optional[str] = json_obj.get("dependency_copy_path", "")
+        self.dependency_origin_path: Optional[str] = json_obj.get("dependency_origin_path", "")
+
+    def to_json(self) -> Dict[str, Any]:
+        """Returns json representation of HubContentDependency object."""
+        json_obj = {att: getattr(self, att) for att in self.__slots__ if hasattr(self, att)}
+        return json_obj
+
+
+class HubContentDescription(JumpStartDataHolderType):
+    """Data class for the Hub Content from session.describe_hub_contents()"""
+
+    __slots__ = [
+        "creation_time",
+        "document_schema_version",
+        "failure_reason",
+        "hub_arn",
+        "hub_content_arn",
+        "hub_content_dependencies",
+        "hub_content_description",
+        "hub_content_display_name",
+        "hub_content_document",
+        "hub_content_markdown",
+        "hub_content_name",
+        "hub_content_search_keywords",
+        "hub_content_status",
+        "hub_content_type",
+        "hub_content_version",
+        "hub_name",
+    ]
+
+    def __init__(self, json_obj: Dict[str, Any]) -> None:
+        """Instantiates HubContentDescription object.
+
+        Args:
+            json_obj (Dict[str, Any]): Dictionary representation of hub content description.
+        """
+        self.from_json(json_obj)
+
+    def from_json(self, json_obj: Dict[str, Any]) -> None:
+        """Sets fields in object based on json.
+
+        Args:
+            json_obj (Dict[str, Any]): Dictionary representation of hub content description.
+        """
+        self.creation_time: int = int(json_obj["creation_time"])
+        self.document_schema_version: str = json_obj["document_schema_version"]
+        self.failure_reason: str = json_obj["failure_reason"]
+        self.hub_arn: str = json_obj["hub_arn"]
+        self.hub_content_arn: str = json_obj["hub_content_arn"]
+        self.hub_content_dependencies: List[HubContentDependency] = [
+            HubContentDependency(dep) for dep in json_obj["hub_content_dependencies"]
+        ]
+        self.hub_content_description: str = json_obj["hub_content_description"]
+        self.hub_content_display_name: str = json_obj["hub_content_display_name"]
+        self.hub_content_document: str = json_obj["hub_content_document"]
+        self.hub_content_markdown: str = json_obj["hub_content_markdown"]
+        self.hub_content_name: str = json_obj["hub_content_name"]
+        self.hub_content_search_keywords: str = json_obj["hub_content_search_keywords"]
+        self.hub_content_status: str = json_obj["hub_content_status"]
+        self.hub_content_type: HubContentType.content_only = json_obj["hub_content_type"]
+        self.hub_content_version: str = json_obj["hub_content_version"]
+        self.hub_name: str = json_obj["hub_name"]
+
+    def to_json(self) -> Dict[str, Any]:
```

Comment:
nit: can this method be inherited from a parent class? Looks the same as `to_json` of `HubDescription`